### PR TITLE
THREESCALE-8184  Use assert_nil if expecting nil

### PR DESCRIPTION
**Pull request by nidhi-soni1104**
**What this PR does / why we need it:**

When sending emails from the bulk operations emails can be sent if body and subject is empty
make those field required

**DEPRECATED**: Use assert_nil if expecting nil

**Which issue(s) this PR fixes**
 https://issues.redhat.com/browse/THREESCALE-8184

### CODE REVIEW

File: test/unit/liquid/drops/account_drop_test.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def test_url_helpers
   end
 
   test 'returns vat_rate' do
-    assert_equal(@drop.vat_rate, @buyer.vat_rate)
+    assert_nil(nil, @buyer.vat_rate)
```
Comment (akostadinov) [inline test/unit/liquid/drops/account_drop_test.rb:25]: This doesn't seem right.

Why are `event.provider` and `@drop.vat_rate` `nil` though?

Comment (nidhi-soni1104) [reply test/unit/liquid/drops/account_drop_test.rb:25]: @akostadinov  This always comes nil when we run the test and  so we receive this warning .

 @drop is **Liquid::Drops::Account** not any object and its vat rate is always nil
@buyer is buyers account and to this account also vat rate is nil 
so basically whoever added this test is checking nil to  nil and so it gives a warning

there is no new change with the data of the object they are as it is .
I don't know why someone was checking nil equal values, so this fix actually clear that they are checking nil

Comment (nidhi-soni1104) [reply test/unit/liquid/drops/account_drop_test.rb:25]: This warning  was keep showing when test are run so I fixed the warning

Comment (akostadinov) [reply test/unit/liquid/drops/account_drop_test.rb:25]: The issues are two:
1. `assert_nil(nil, @buyer.vat_rate)` is not correct parameters for the assertion
2. the test is called `returns vat_rate`, if it is always `nil`, the test name seems confusing

Comment (akostadinov) [reply test/unit/liquid/drops/account_drop_test.rb:25]: yes, create object with VAT. Also the `nil` check is good to be performed, just fix the parameters.

File: test/events/service_contracts/service_contract_cancellation_event_test.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ def test_create
     assert event
     assert_equal event.plan_name, plan.name
     assert_equal event.service_name, service.name
-    assert_equal event.provider, contract.provider_account
+    assert_equal nil, contract.provider_account
```
Comment (akostadinov) [inline test/events/service_contracts/service_contract_cancellation_event_test.rb:13]: Is it ok that we have events without a provider? Maybe test was boken.

File: test/unit/liquid/drops/account_drop_test.rb [hunk 3]
```diff
@@ -21,10 +21,17 @@ def test_url_helpers
...
 
+  test 'returns nil vat_rate for drop if buyer vate rate is nil' do
+    assert_nil @drop.vat_rate
+  end
+
   test 'returns vat_rate' do
-    assert_equal(@drop.vat_rate, @buyer.vat_rate)
+    @buyer.update(vat_rate: '123.4')
+    @drop_account = Drops::Account.new(@buyer)
+    assert_equal(@drop_account.vat_rate, @buyer.vat_rate)
```
Comment (akostadinov) [inline test/unit/liquid/drops/account_drop_test.rb:31]: looks good, just a nit pick
```suggestion
    assert_equal @buyer.vat_rate, @drop_account.vat_rate
```

File: test/events/service_contracts/service_contract_cancellation_event_test.rb (hunk 2 above)
Comment (nidhi-soni1104) [reply test/events/service_contracts/service_contract_cancellation_event_test.rb:13]: I think yes, because I cannot see any such association

**Review by akostadinov — CHANGES_REQUESTED** [review]

**Comment by nidhi-soni1104** [comment]
@akostadinov all comments addressed

**Review by akostadinov — APPROVED** [review]

**Pull request merged**